puzzles/views.py

```diff
@@ -77,6 +80,42 @@ def slack_guess(request):
     return HttpResponse("Answer " + answer_text + " has been submitted!")
 
 
+@csrf_exempt
+def slack_events(request):
+    if request.method != 'POST':
+        return HttpResponse("POST request expected")
+
+    event = json.loads(request.body)
+    token = event.get('token')
+    # TODO(erwa): Move SLACK_VERIFICATION_TOKEN into settings.py
+    if token != os.environ.get("SLACK_VERIFICATION_TOKEN"):
+        return HttpResponseForbidden()
+
+    # one time events API verification
+    challenge = event.get('challenge')
+    if challenge:
+        return HttpResponse(challenge)
+
+    event = event.get('event')
+    print('Received Slack event:', event)
+
+    event_type = event.get('type')
+    if event_type == 'member_joined_channel':
+        email = SlackClient.getInstance().get_user_email(event.get('user'))
+        user = Puzzler.objects.get(email=email)
```

Comment:
IMO it's fine to not track these people -- probably when their puzzler object gets created, we can check if that user exists in the existing puzzle channels.

Reply:
Yeah I'm not worried about that, but what happens to this code? Slack is okay if we don't send back a success code?

Reply:
We will send back a success code in this case -- the HttpResponse at the end of this method.

Yeah, there's some edge cases this initial code isn't handling right now. Also not sure how long Slack will buffer events or resend them in the case that Small Board is down. But anyways, active users is a heuristic anyways, since people could be in a Slack channel but not actively working on the puzzle.

I'll also look into freezing a history of people who have worked on a puzzle once a puzzle has been marked SOLVED.

Reply:
I think Puzzler.objects.get(email=email) will throw an exception if the email doesn't exist yet. It's fine as long as slack doesn't continue to spam us with the same event lol

Reply:
Ah, thanks for pointing that out. I'll add a try-except to handle this.

Reply:
Added a try-except. Also, Slack will resend "failed" events up to 3 times before giving up: https://api.slack.com/events-api#errors

Reply:
The error message in the case when a user doesn't exist looks like:
```
User with email erwaman@gmail.com not found. Exception: Puzzler matching query does not exist.
```